File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLUpdateChatAdmins.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.TLContext;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readInt;
import static com.github.badoualy.telegram.tl.StreamUtils.readTLBool;
import static com.github.badoualy.telegram.tl.StreamUtils.writeInt;
import static com.github.badoualy.telegram.tl.StreamUtils.writeTLBool;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLUpdateChatAdmins extends TLAbsUpdate {
    public static final int CLASS_ID = 0x6e947941;

    protected int chatId;

    protected boolean enabled;

    protected int version;

    public TLUpdateChatAdmins() {
    }

    public TLUpdateChatAdmins(int chatId, boolean enabled, int version) {
        this.chatId = chatId;
        this.enabled = enabled;
        this.version = version;
    }

    @Override
    public void serializeBody(OutputStream stream) throws IOException {
        writeInt(chatId, stream);
        writeTLBool(enabled, stream);
        writeInt(version, stream);
    }

    @Override
    @SuppressWarnings("unchecked")
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {
        chatId = readInt(stream);
        enabled = readTLBool(stream);
        version = readInt(stream);
    }

    @Override
    public String toString() {
        return "updateChatAdmins#6e947941";
    }

    @Override
    public int getClassId() {
        return CLASS_ID;
    }

    public int getChatId() {
        return chatId;
    }

    public void setChatId(int chatId) {
        this.chatId = chatId;
    }

    public boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public int getVersion() {
        return version;
    }

    public void setVersion(int version) {
        this.version = version;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLUpdateChatAdmins.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException

/**
 * updateChatAdmins#6e947941
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLUpdateChatAdmins() : TLAbsUpdate() {
    var chatId: Int = 0

    var enabled: Boolean = false

    var version: Int = 0

    private val _constructor: String = "updateChatAdmins#6e947941"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(
            chatId: Int,
            enabled: Boolean,
            version: Int
    ) : this() {
        this.chatId = chatId
        this.enabled = enabled
        this.version = version
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeInt(chatId)
        writeBoolean(enabled)
        writeInt(version)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        chatId = readInt()
        enabled = readBoolean()
        version = readInt()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += SIZE_INT32
        size += SIZE_BOOLEAN
        size += SIZE_INT32
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLUpdateChatAdmins) return false
        if (other === this) return true

        return chatId == other.chatId
                && enabled == other.enabled
                && version == other.version
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x6e947941.toInt()
    }
}
